File: backend/Services/Parsers/SedecordleParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class SedecordleParser : BasicScoreResultParser
    {
        private ILogger<SedecordleParser> _logger;

        public SedecordleParser(ILogger<SedecordleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Sedecordle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"Daily #[^\n]+\n(?<{ScoreGroup}>.*?)\n{ExtraContent}", RegexOptions.Singleline);
        protected override string? ExtraContent => "sedecordle.com\n#sedecordle";
        public override string Url => "https://www.sedecordle.com/?mode=daily";

        public override string? GetScoreValue(DailyResult dailyResult)
        {
            if (dailyResult.Scores == null)
            {
                return null;
            }

            return "\"" + string.Join(",", Enumerable.Range(0, 16).ToArray()
                .Select((i) => {
                    if (dailyResult.Scores.Count <= i)
                    {
                        return "X";
                    }

                    return dailyResult.Scores[i].ToString();
                })) + "\"";
        }

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            var scoresText = parserResults.Groups[ScoreGroup].Value;

            // 0\uFE0F\u20E3 = 0️⃣
            // 1\uFE0F\u20E3 = 1️⃣
            // 2\uFE0F\u20E3 = 2️⃣
            // 3\uFE0F\u20E3 = 3️⃣
            // 4\uFE0F\u20E3 = 4️⃣
            // 5\uFE0F\u20E3 = 5️⃣
            // 6\uFE0F\u20E3 = 6️⃣
            // 7\uFE0F\u20E3 = 7️⃣
            // 8\uFE0F\u20E3 = 8️⃣
            // 9\uFE0F\u20E3 = 9️⃣
            var scores = Regex.Matches(scoresText, @"[\d\uFE0F\u20E3]+")
                .OfType<Match>()
                .Select(match => {
                    var value = match.Value
                        .Replace("\uFE0F", string.Empty)
                        .Replace("\u20E3", string.Empty);
                    return int.Parse(value);
                })
                .ToList();

            dailyResult.Scores = scores;

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/sedecordle-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class SedecordleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Sedecordle';
  readonly helpText = null;
  readonly url = 'https://www.sedecordle.com/?mode=daily';
  protected readonly extraContent = 'sedecordle.com\n#sedecordle';
  
  // Matches "Daily #1374\n1️⃣8️⃣⬛0️⃣4️⃣..." - captures the score block
  // C# regex: `Daily #[^\n]+\n(?<score>.*?)\n{ExtraContent}` with Singleline flag
  protected readonly parser = new RegExp(
    `Daily #[^\\n]+\\n(?<${BasicScoreResultParser.SCORE_GROUP}>.*?)\\nsedecordle.com\\n#sedecordle`,
    's'
  );
  
  override getScoreValue(dailyResult: DailyResult): string | null {
    if (!dailyResult.scores) return null;
    
    const values = Array.from({ length: 16 }, (_, i) => {
      if (dailyResult.scores && dailyResult.scores.length > i) {
        return dailyResult.scores[i].toString();
      }
      return 'X';
    });
    
    return `"${values.join(',')}"`;
  }
  
  protected override setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (!hasGroup(match, BasicScoreResultParser.SCORE_GROUP)) {
      return dailyResult;
    }
    
    const scoresText = getGroup(match, BasicScoreResultParser.SCORE_GROUP) || '';
    
    // Parse emoji digits: 0️⃣ through 9️⃣
    // These are composed of digit + \uFE0F + \u20E3
    const digitMatches = scoresText.matchAll(/[\d\uFE0F\u20E3]+/g);
    const scores: number[] = [];
    
    for (const digitMatch of digitMatches) {
      const value = digitMatch[0]
        .replace(/\uFE0F/g, '')
        .replace(/\u20E3/g, '');
      const score = parseInt(value, 10);
      if (!isNaN(score)) {
        scores.push(score);
      }
    }
    
    if (scores.length > 0) {
      dailyResult.scores = scores;
    }
    
    return dailyResult;
  }
}
